contracts/src/libraries/Transcript.sol

```diff
@@ -39,39 +37,24 @@ library Transcript {
         internal
         pure
     {
-        self.transcript = abi.encodePacked(self.transcript, BN254.g1Serialize(comm));
+        self.transcript = abi.encodePacked(self.transcript, comm.x, comm.y);
     }
 
     function getAndAppendChallenge(TranscriptData memory self) internal pure returns (uint256) {
-        bytes32 hash;
+        bytes memory transcript = self.transcript;
+        uint256 ret = uint256(keccak256(transcript)) % BN254.R_MOD;
 
-        bytes32 a = self.state;
-        bytes memory b = self.transcript;
-
-        // Computes keccak256(bytes32 a, bytes b)
         assembly {
-            // Load the length of 'b'
-            let bLength := mload(b)
-
-            // Allocate memory for 'a' + 'b'
-            let data := mload(0x40) // Load free memory pointer
-
-            // Store 'a' in memory
-            mstore(data, a)
-
-            // Copy 'self.transcript' to memory after 'self.state'
-            let dataOffset := add(data, 32) // Start right after 'self.state'
-            for { let i := 0 } lt(i, bLength) { i := add(i, 0x20) } {
-                mstore(add(dataOffset, i), mload(add(add(b, i), 0x20)))
-            }
-
-            // Compute the keccak256 hash of the data
-            hash := keccak256(data, add(32, bLength))
+            let len := mload(transcript)
```

Comment:
Suggestion: add a comment explaining what is being done in the assembly code, e,g.:

`// self.transcript = abi.encodePacked(self.transcript,ret);`

Reply:
will do, btw, if we replace with your line, the gas will jump up to:
```
LightClient_newFinalizedState_Test:testCorrectUpdateBench() (gas: 696716)
PlonkVerifier_verify_Test:test_verify_succeeds() (gas: 531948)
```

so our assembly did save gas!

Reply:
Definitely worth the effort!

Reply:
done in [5fba575](https://github.com/EspressoSystems/espresso-sequencer/pull/1801/commits/5fba575b75e47783b5c12d294169ed96dfe2787c)